batch/batch/utils.py

```diff
@@ -115,19 +115,34 @@ def __repr__(self):
         return f'global {self._global_counter}'
 
 
-async def query_billing_projects(db, user=None, billing_project=None):
-    args = []
+def users_record_to_dict(record):
+    if record['users'] is None:
+        record['users'] = []
+    else:
+        record['users'] = json.loads(record['users'])
+    return record
 
-    where_conditions = ["billing_projects.`status` != 'deleted'"]
+
+def _query_billing_projects_conditions(
+    user: Optional[str], billing_project: Optional[str]
+) -> Tuple[List[str], List[str]]:
+    args = []
+    conditions = ["billing_projects.`status` != 'deleted'"]
 
     if user:
-        where_conditions.append("JSON_CONTAINS(users, JSON_QUOTE(%s))")
+        conditions.append("JSON_CONTAINS(users, JSON_QUOTE(%s))")
         args.append(user)
 
     if billing_project:
-        where_conditions.append('billing_projects.name_cs = %s')
+        conditions.append('billing_projects.name_cs = %s')
         args.append(billing_project)
```

Comment:
While this deduplicates some code, it also fragments the query, and I can't really understand what this function is doing unless I look at where it's used (where does this users json come from?). I think it's worth sacrificing the duplication to put this code in place in the functions that call this one instead of having this function.